projects/aca-content/src/lib/components/search/search-input/search-input.component.spec.ts

```diff
@@ -1,62 +1,116 @@
       imports: [AppTestingModule, ReactiveFormsModule, SearchInputComponent, SearchInputControlComponent],
-      schemas: [NO_ERRORS_SCHEMA]  
+      providers: [{ provide: Store, useValue: storeSpy }],
+      schemas: [NO_ERRORS_SCHEMA]
     }).compileComponents();
 
     fixture = TestBed.createComponent(SearchInputComponent);
     component = fixture.componentInstance;
+    store = TestBed.inject(Store) as jasmine.SpyObj<Store<AppStore>>;
+    store.pipe.and.returnValue(of([]));
+    fixture.detectChanges();
+  });
+
+  function getFirstError(): string {
+    const error = fixture.debugElement.query(By.directive(MatError));
+    return error?.nativeElement.textContent.trim();
+  }
 
-    // (component as any).searchInputControl = mockControl;
+  function openSearchContainer(): void {
+    const menuButton = fixture.debugElement.query(By.css('.app-search-container'));
+    menuButton?.nativeElement.click();
     fixture.detectChanges();
+  }
+
+  it('should show required error when field is empty and touched', () => {
+    openSearchContainer();
+    component.searchInputControl.searchFieldFormControl.setValue('');
+    component.searchInputControl.searchFieldFormControl.markAsTouched();
+    fixture.detectChanges();
+    expect(getFirstError()).toBe('SEARCH.INPUT.REQUIRED');
   });
 
-  describe('Validation Behavior', () => {
-    function getFirstError(): string {
-      const error = fixture.debugElement.query(By.directive(MatError));
-      return error?.nativeElement.textContent.trim();
-    }
-
-    it('should show required error when field is empty and touched', () => {
-      component.searchInputControl.searchFieldFormControl.setValue('');
-      component.searchInputControl.searchFieldFormControl.markAsTouched();
-      fixture.detectChanges();
-
-      const error = getFirstError();
-      fixture.detectChanges();
-      expect(error).toBe('SEARCH.INPUT.REQUIRED');
-    });
-
-    // it('should not show error when field has value', () => {
-    //   mockControl.searchFieldFormControl.setValue('test');
-    //   mockControl.searchFieldFormControl.markAsTouched();
-    //   fixture.detectChanges();
-
-    //   const error = fixture.debugElement.query(By.directive(MatError));
-    //   expect(error).toBeNull();
-    // });
-
-    // it('should not show error when field is untouched', () => {
-    //   mockControl.searchFieldFormControl.setValue('');
-    //   mockControl.searchFieldFormControl.markAsUntouched();
-    //   fixture.detectChanges();
-
-    //   const error = fixture.debugElement.query(By.directive(MatError));
-    //   expect(error).toBeNull();
-    // });
+  it('should not show error when field has value', () => {
+    openSearchContainer();
+    component.searchInputControl.searchFieldFormControl.setValue('test');
+    component.searchInputControl.searchFieldFormControl.markAsTouched();
+    fixture.detectChanges();
+    const error = fixture.debugElement.query(By.directive(MatError));
```

Comment:
instead of using directly debugElement, please try to use UnitTestingUtils helper